llama_stack/apis/datasets/datasets.py

```diff
@@ -4,19 +4,96 @@
 # This source code is licensed under the terms described in the LICENSE file in
 # the root directory of this source tree.
 
-from typing import Any, Dict, List, Literal, Optional, Protocol
+from enum import Enum
+from typing import Annotated, Any, Dict, List, Literal, Optional, Protocol, Union
 
 from pydantic import BaseModel, Field
 
-from llama_stack.apis.common.content_types import URL
-from llama_stack.apis.common.type_system import ParamType
 from llama_stack.apis.resource import Resource, ResourceType
-from llama_stack.schema_utils import json_schema_type, webmethod
+from llama_stack.schema_utils import json_schema_type, register_schema, webmethod
+
+
+class DatasetPurpose(Enum):
+    """
+    Purpose of the dataset. Each type has a different column format.
+    :cvar post-training/messages: The dataset contains messages used for post-training. Examples:
+        {
+            "messages": [
+                {"role": "user", "content": "Hello, world!"},
+                {"role": "assistant", "content": "Hello, world!"},
+            ]
+        }
+    """
+
+    post_training_messages = "post-training/messages"
```

Comment:
Keeping it post_training for now to be consistent. Will need to migrate altogether if we want to change post_training to tune.